src/OrleansTelemetryConsumers.DataDog/DataDogTelemetryConsumer.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Newtonsoft.Json;
+using Orleans.Runtime;
+using StatsdClient;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Orleans.TelemetryConsumers.DataDog
+{
+    public class DataDogTelemetryConsumer : IMetricTelemetryConsumer, IEventTelemetryConsumer
+    {
+        public DataDogTelemetryConsumer(string statsdServerName = "127.0.0.1", int statsdPort = 8125, string prefix = "")
+        {
+            var dogstatsdConfig = new StatsdConfig
+            {
+                StatsdServerName = statsdServerName,
+                StatsdPort = statsdPort,
+                Prefix = prefix
+            };
+
+            DogStatsd.Configure(dogstatsdConfig);
+        }
+
+        public void DecrementMetric(string name)
+        {
+            DogStatsd.Decrement(name);
+        }
+
+        public void DecrementMetric(string name, double value)
+        {
+            DogStatsd.Decrement(name, (int)value);
+        }
+
+        public void IncrementMetric(string name)
+        {
+            DogStatsd.Increment(name);
+        }
+
+        public void IncrementMetric(string name, double value)
+        {
+            DogStatsd.Increment(name, (int)value);
+        }
+
+        public void TrackMetric(string name, TimeSpan value, IDictionary<string, string> properties = null)
+        {
+            DogStatsd.Increment(name, (int)value.TotalMilliseconds, tags: ToStringArray(properties));
+        }
+
+        public void TrackMetric(string name, double value, IDictionary<string, string> properties = null)
+        {
+            DogStatsd.Increment(name, (int)value, tags: ToStringArray(properties));
```

Comment:
Oh, I see. But can we be sure the range is valid in casts? If we are sure, I think it'd be reassuring to have it in comments. Always leaving out fractional parts doesn't seem a right thing to do. I don't have a clear idea how much it would matter during operations, but longer term statistics would compound the effect. Would banker's rounding and then casting introduce too much overhead?